zeebe/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/AdHocSubProcessValidator.java

```diff
@@ -57,6 +64,39 @@ public void validate(
     }
   }
 
+  private static void validateTaskDefinition(
+      final AdHocSubProcess adHocSubProcess,
+      final ValidationResultCollector validationResultCollector) {
+    final ZeebeAdHocImplementationType implementationType =
+        Optional.ofNullable(adHocSubProcess.getSingleExtensionElement(ZeebeAdHoc.class))
+            .map(ZeebeAdHoc::getImplementationType)
+            .orElse(ZeebeAdHocImplementationType.BPMN);
+
+    final Collection<ZeebeTaskDefinition> taskDefinitions =
+        Optional.ofNullable(adHocSubProcess.getExtensionElements())
+            .map(
+                extensionElements ->
+                    extensionElements.getChildElementsByType(ZeebeTaskDefinition.class))
+            .orElse(Collections.emptyList());
+
+    if (implementationType == ZeebeAdHocImplementationType.BPMN && !taskDefinitions.isEmpty()) {
+      validationResultCollector.addError(
+          0,
+          String.format(
+              "Must not have a zeebe:%s extension element with implementation type %s.",
+              ZeebeConstants.ELEMENT_TASK_DEFINITION, ZeebeAdHocImplementationType.BPMN));
+    }
```

Comment:
🤔 Not 100% sure if this makes sense or if this should be just allowed (and ignored).